src/migration/service/migrator.ts

```diff
@@ -57,3 +85,34 @@ export async function resolveSqlMigrations(migrationPath: string): Promise<SqlMi
 
   return Promise.all(migrationPromises);
 }
+
+/**
+ * Resolve all the migration source for each of the migration entries using the configurations.
+ *
+ * @param {string} migrationPath
+ * @param {string} extension
+ * @returns {Promise<SqlMigrationEntry[]>}
+ */
+export async function resolveJavaScriptMigrations(
+  migrationPath: string,
+  extension: string = FileExtensions.JS
+): Promise<JavaScriptMigrationEntry[]> {
+  const migrationNames = await getJavaScriptMigrationNames(migrationPath, extension);
+  const migrationPromises = migrationNames.map(async name => {
+    const filename = `${name}.${extension}`;
+
+    // transpile & execute file required on the fly
+    require('ts-node').register({
+      transpileOnly: true
+    });
```

Comment:
Do it only for typescript files. It's not needed for all the js files.

Reply:
Also, do you need to run it in the loop here? Shouldn't it just work if you move it outside the map callback?

Reply:
@samirsilwal Yes, you can check the extension and require it outside the loop.

Reply:
ok will make  the changes